addons/hr_holidays/static/src/js/time_off_calendar.js

```diff
@@ -117,34 +117,42 @@ odoo.define('hr_holidays.dashboard.view_custo', function(require) {
          *
          * @private
          */
-        _onNewAllocation: function () {
-            let self = this;
-
+        _onNewAllocation: async function () {
             let domain = [
                 ['name', '=', 'hr.leave.view.form.dashboard'],
                 ['model', '=', 'hr.leave.allocation']
             ];
 
-            self._rpc({
+            const ids = await this._rpc({
                 model: 'ir.ui.view',
                 method: 'search',
                 args: [domain],
-            }).then(function(ids) {
-                self.allocationDialog = new dialogs.FormViewDialog(self, {
-                    res_model: "hr.leave.allocation",
-                    view_id: ids,
-                    context: {
-                        'default_employee_ids': self.context.employee_id,
-                        'default_state': 'confirm',
-                    },
-                    title: _t("New Allocation"),
-                    disable_multiple_selection: true,
-                    on_saved: function() {
-                        self.reload();
-                    },
+            });
+
+            let default_employee_ids = this.context.employee_id;
+
+            if (!default_employee_ids) {
+                default_employee_ids = await this._rpc({
+                    model: 'hr.employee',
+                    method: 'search',
+                    args: [[['user_id', '=', this.context.uid]]]
                 });
-                self.allocationDialog.open();
+            }
+
+            this.allocationDialog = new dialogs.FormViewDialog(this, {
+                res_model: "hr.leave.allocation",
+                view_id: ids,
+                context: {
+                    'default_employee_ids': default_employee_ids,
```

Comment:
Thinking about it... The default_employee_ids is only needed when coming from the employee form and clicking on the time off smart button as it is done in time_off_calendar_employee.js:
https://github.com/odoo/odoo/blob/38e5d002acc2e516987acc0afe38648473fff708/addons/hr_holidays/static/src/js/time_off_calendar_employee.js#L232
So removing the default_employee_ids in the _onNewAllocation method here fixes it :)

```suggestion
                    
```

Reply:
Indeed, you're right. I'll do that then